### DISCUSSION TITLE
Loading Pre-segmented data into Orthanc the proper way so Monailabel can use it

### DISCUSSION THREAD

**Discussion by cliff-lindsay** [Project-MONAI/MONAILabel#1371] [Q&A]
Dear MonaiLabel Community,

I was wondering if you could help; I would like to use MonaiLabel radiology app in conjunction with Orthanc and OHIF. I have the MonaiLabel server up and running and can load images in OHIF. The problem I'm encountering is that I would like to populate Orthanc with with the segmentations of the image that were performed previously but I cannot get Monailabel to recognize the previously generated labels (generated outside of MoaniaLabel).

The segmentation files are proper DICOM images and conform to the DICOM-SEG standard. Furthermore, they are associated with the medical images (CTs) in that they are listed in the same study as their respective CTs (see screen shot below).  The DICOM-SEGs are single 3d volumetric dicom files.

I was wondering if there is some guidance as to how to get pre-segmented images to work with MonaiLabel from within Orthanc. I have seen multiple examples and a variety of explanations on how to do this with 3D slicer and saving the images in directories, but I would really like to integrate MonaiLabel into our Orthanc PACS workflow if possible without having to re-segment 100s of images. Seems like there might be some additional setup required on MonaiLabel's side that I would have to do to make this work. Any help would be really appreciated. I have combining through the docs and discussions and have watched multiple videos to try to figure this out.

![explorer2_example](https://user-images.githubusercontent.com/84593952/229218714-1727a864-972a-4604-aa07-258f90e431ed.png)

**Comment by SachidanandAlle** [comment]
monailabel considers existing DICOM segs for Orthanc.. there must be something missing piece of information to understand...
possibly it was existing those SEG to be correctly tagged against the Series.. it looks for `ReferencedSeriesSequence` in the meta info for SEG...

https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/dicom.py#L146

You can share/go through the monailabel logs when you have started the server.
You can clean cache (possibly in ~/.cache/monailabel) and run again.. 

Are you seeing any warnings in the log.. something like: https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/dicom.py#L154

**Comment by diazandr3s** [comment]
Hi @cliff-lindsay,

We use the [**_dicomweb-client_**](https://github.com/dcmjs-org/dicomweb-client) library to communicate with Orthanc.
This is the [datastore class](https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/dicom.py#L42) MONAI Label uses to communicate with Orthanc (DICOMWeb API).
You may want to load the DICOM-SEG files associate to each volume before triggering MONAI Label training.

Hope this help
